### DISCUSSION TITLE
useWatch not workwell for children unregister

### DISCUSSION THREAD

**Discussion by nagq** [react-hook-form/react-hook-form#8709] [Q&A]
DEMO:  https://codesandbox.io/s/react-hook-form-control-forked-c0zoyx

![image](https://user-images.githubusercontent.com/15318124/180208003-a7962704-8c9b-41a7-b7e1-7b135bd22fa2.png)

**Comment by Moshyfawn** [comment]
It's the way React's component lifecycle works, not a RHF issue.

The parent component doesn't get re-rendered due to child component changing state. To solve this you need to lift your state up.

Here's [a basic example](https://codesandbox.io/s/react-hook-form-control-forked-irig2g?file=%2Fsrc%2FApp.tsx)

**Reply by remziatay** [reply]
Can't RHF run `unregister` immediately in ref function instead of in `_removeUnmounted`? That would trigger a render in Parent component.
Example workaround to simulate what I mean:

```jsx
  const { ref, ...rest } = register("name");

  const newRef = React.useCallback((el) => {
    ref(el);
    if (!el) {
      unregister("name");
    }
  }, []);

  return (
    <div>
      {show && (
        <div>
          <input ref={newRef} {...rest} />
        </div>
      )}
    </div>
  );

```

**Reply by bluebill1049** [reply]
not possible with native input as ref call back provided only `null`
